.github/workflows/build-test.yaml

```diff
@@ -340,6 +340,71 @@ jobs:
           docker push ttl.sh/automated-${{ github.run_id }}/postgres:${{ steps.dotenv.outputs.POSTGRES_14_TAG }}
 
 
+  # only run validate-kurl-addon if changes to "deploy/kurl/**"
+  kurl-addon-changes-filter:
+    runs-on: ubuntu-20.04
+    needs: [ can-run-ci, enable-tests ]
+    outputs:
+      ok-to-test: ${{ steps.filter.outputs.kurl-addon }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            kurl-addon:
+              - 'deploy/kurl/**'
+  validate-kurl-addon:
+    runs-on: ubuntu-20.04
+    if: ${{ needs.kurl-addon-changes-filter.outputs.ok-to-test == 'true' }}
+    needs: [ can-run-ci, enable-tests, generate-tag, kurl-addon-changes-filter, build-kots, build-kurl-proxy, build-migrations ]
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: set outputs
+        id: vars
+        run: |
+          addon_version=${{ needs.generate-tag.outputs.tag }}
+          echo "::set-output name=addon_version::${addon_version#v}"
+      - name: download kots binary
+        uses: actions/download-artifact@v2
+        with:
+          name: kots
+          path: bin/
+      - name: prepare kots binary executable
+        run: |
+          chmod +x bin/*
+          tar -C bin/ -czvf bin/kots.tar.gz kots
+      - name: push dex image
```

Comment:
nit pick, this could be moved to a separate job like the push-postgres and push-minio jobs: https://github.com/replicatedhq/kots/blob/main/.github/workflows/build-test.yaml#L299